lib/wallaroo/core/sink/connector_sink/connector_sink.pony

```diff
@@ -622,17 +671,28 @@ actor ConnectorSink is Sink
     """
     2nd-half logic for barrier_fully_acked().
     """
-    let queued = _phase.queued()
-    _use_normal_processor()
-    for q in queued.values() do
-      match q
-      | let qm: QueuedMessage =>
-        qm.process_message(this)
-      | let qb: QueuedBarrier =>
-        qb.inject_barrier(this)
+    match _phase
```

Comment:
We shouldn't be matching on phases, but delegating to them.

Reply:
Fixed by d0f9c924dac